tex/generic/pgf/frontendlayer/tikz/tikz.code.tex

```diff
@@ -2580,28 +2580,64 @@
 
 \def\tikz@fchar oreach{\tikz@foreach}%
 
-\def\tikz@foreach{%
+\def\tikzforeach@smugglers@cove{}
+\let\tikztostart=\relax
```

Comment:
I had a look at `tikzlibrarytopaths.code.tex` and while that is where `\tikztostart` is _used_, it is never defined there.  The definitions are contained in the core TikZ parser in that whenever TikZ encounters a coordinate then it sets `\tikztostart` to something sensible so that _if_ the next command is a `to` path then the start is set up correctly.  Similarly with `\tikz@tangent` (which is mainly used in the `calc` library).

So I'm happy to move it, but just want to double-check first.